core/domain/suggestion_services.py

```diff
@@ -216,7 +215,7 @@ def get_all_stale_suggestion_ids():
     )
 
 
-def _update_suggestion(suggestion):
+def update_suggestion(suggestion):
```

Comment:
Why are we changing this private function to a public function? As an endpoint for suggestion_services, we only allow a few specific updates on the suggestion model like accept/reject etc. and we have a specific method for these. As we need to perform specific tasks for different kinda updates on the suggestion (like send emails for if accept suggestion), right?

Reply:
But it's intentionally made private, so that sugestion_services can be called with a specific public method which can internally use the _update method, right?

Reply:
> I think it's cleaner to call update_suggestion in the migration since I'm setting the language_code field by updating the domain object. 

I agree that it will look cleaner using this method but unfortunately, this method is private.

> This approach is also taken in the ExplorationMigrationJobManger class (since the update function is public for explorations)
Exploration is a bit different than the suggestion, one can change all the properties of exploration were as we allow updating few parts of the suggestion model. the update method in exploration doesn't take the update exploration as a param, basically, it handles all the changes (Apply change list etc.) which isn't true for the suggestion. I think the name of the function is a bit different in the case of suggestion_services (I was expecting it to be _save_suggestion or something around that!)